bmq-sdk/src/main/java/com/bloomberg/bmq/QueueOptions.java

```diff
@@ -314,13 +345,16 @@ public Builder merge(QueueOptions options) {
             if (options.hasSuspendsOnBadHostHealth()) {
                 setSuspendsOnBadHostHealth(options.getSuspendsOnBadHostHealth());
             }
+
```

Comment:
So before subscriptions we set fields which have been set in 'options'.
Current logic with subscription always override current value which seems to be a bit inconsistent with the logic above.

It looks like we need to use `Optional` here as well

Reply:
For subscriptions `List`, its an overkill to use `Optional`, when we can check `List` length.
I also think that adding these 'has<FieldName>()' methods was a design mistake.
Would rather check for length > 0 before overriding.

Reply:
The 'has' things was introduced when there was a work host health monitoring.
Anyway, if we break the behaviour, we introduce inconsistent with other fields.
Probably this could be discussed.